src/udp/connection_cookie.rs

```diff
@@ -0,0 +1,214 @@
+        // seed: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
+
+        const ID_STABLE: i64 = 8313098377115836164;
+        const ID_NIGHTLY: i64 = -4342702349221980886;
+
+        let test_cookie = make_test_cookie(None);
+        assert!((test_cookie == ConnectionId(ID_STABLE) || (test_cookie == ConnectionId(ID_NIGHTLY))))
+    }
+
+    #[test]
+    fn it_should_make_different_connection_cookie_with_different_remote_addresses() {
+        let test_remote_address_1 = SocketAddr::new(IpAddr::V6(Ipv6Addr::UNSPECIFIED), 1);
+        let test_remote_address_2 = SocketAddr::new(IpAddr::V6(Ipv6Addr::UNSPECIFIED), 2);
+        let test_remote_address_3 = SocketAddr::new(IpAddr::V6(Ipv6Addr::LOCALHOST), 1);
+
+        assert_ne!(
+            make_test_cookie(Some(&test_remote_address_1)),
+            make_test_cookie(Some(&test_remote_address_2))
+        );
+
+        assert_ne!(
+            make_test_cookie(Some(&test_remote_address_1)),
+            make_test_cookie(Some(&test_remote_address_3))
+        );
+
+        assert_ne!(
+            make_test_cookie(Some(&test_remote_address_2)),
+            make_test_cookie(Some(&test_remote_address_3))
+        )
+    }
+
+    #[test]
+    fn it_should_make_different_cookies_for_the_next_period() {
+        let cookie_now = make_test_cookie(None);
+
+        StoppedClock::local_add(&get_period_length()).unwrap();
+
+        let cookie_next = make_test_cookie(None);
+
+        assert_ne!(cookie_now, cookie_next)
+    }
+
+    #[test]
+    fn it_cookies_should_be_valid_for_this_and_next_periods() {
+        let cookie_now = make_test_cookie(None);
+
+        check_connection_cookie(&get_test_socket_addr(), &cookie_now).unwrap();
+
+        StoppedClock::local_add(&get_period_length()).unwrap();
+
+        check_connection_cookie(&get_test_socket_addr(), &cookie_now).unwrap();
+    }
+
+    #[test]
+    fn it_cookies_should_be_valid_for_the_last_period() {
+        let cookie_now = make_test_cookie(None);
+
+        let last_period = get_last_valid_period_now();
+
+        StoppedClock::local_set(&detail::get_time_at_period_start(&(last_period)));
+
+        check_connection_cookie(&get_test_socket_addr(), &cookie_now).unwrap();
+    }
+
+    #[test]
+    #[should_panic]
+    fn it_cookies_should_be_not_valid_after_their_periods_has_expired() {
+        let cookie_now = make_test_cookie(None);
+
+        let last_period = get_last_valid_period_now();
+
+        StoppedClock::local_set(&detail::get_time_at_period_end(&(last_period)));
+
+        check_connection_cookie(&get_test_socket_addr(), &cookie_now).unwrap();
+    }
+}
+
+mod detail {
+    use std::collections::hash_map::DefaultHasher;
+    use std::hash::{Hash, Hasher};
+    use std::time::Duration;
+
+    use super::*;
+    use crate::protocol::clock::{DefaultClock, Time};
+    use crate::protocol::crypto::keys::seeds::{DefaultSeed, Seed};
+
+    pub(crate) type Period = u128;
```

Comment:
@da2ce7 I think the name is too generic, but I do not come up with a better name. Maybe `FixedPeriod`. I think we need to add something to make it clear period starts at an exact minute. And they start counting at Unix Epoch.

- `FixedPeriod`
- `UnixEpochPeriod`

Reply:
@josecelano Thanks, and done!